lib/pages/getStaticProps/redirects.js

```diff
@@ -15,22 +41,105 @@ pages.forEach(({ route, dataRoute, srcRoute }) => {
   const conditions = ["Cookie=__prerender_bypass,__next_preview_data"];
   const target = `/.netlify/functions/${functionName}`;
 
-  // Add one redirect for the page, but only when the NextJS
-  // preview mode cookies are present
-  redirects.push({
-    route,
-    target,
-    force: true,
-    conditions,
-  });
-
-  // Add one redirect for the data route, same conditions
-  redirects.push({
-    route: dataRoute,
-    target,
-    force: true,
-    conditions,
-  });
+  if (nextConfig.i18n) {
+    const { defaultLocale, locales = [] } = nextConfig.i18n;
+    const isDynamic = !!srcRoute;
+    if (!isDynamic) {
+      // Add a redirect for "expected" route to the defaultLocale-prepended route
+      // i.e. /getStaticProps -> /en/getStaticProps
+      if (defaultLocale) {
+        redirects.push({
+          route,
+          target: `/${defaultLocale}${route}`,
+        });
+      }
+      // Add a preview mode redirect for every locale
+      locales.forEach((locale) => {
+        const pageRoute = `/${locale}${route}`;
+        redirects.push({
+          route: pageRoute,
+          target,
+          force: true,
+          conditions,
+        });
+        redirects.push({
+          route: getDataRouteForRoute(route, locale),
+          target,
+          force: true,
+          conditions,
+        });
+      });
```

Comment:
Do we not need a preview mode redirect for the "naked" URL, too? I.e., the URL without locale prefix, such as `/my-preview-page`?

Reply:
hm good question. the preview mode cypress tests pass for naked urls so i think it's fine but i'm not sure why lol. will investigate

Reply:
you were right. my fix is not great but wasnt really a better option https://github.com/netlify/next-on-netlify/pull/75/commits/96959bd3204a2166575e5f880dc9f93c2221b534